samples/boards/numaker/system_off/src/main.c

```diff
@@ -0,0 +1,65 @@
+/*
```

Comment:
Why is this introduced as a board specific sample? Did I miss something preventing this one to be generic?

Reply:
it's similar to \zephyr\samples\boards\stm32\power_mgmt\serial_wakeup\src.
And current sys_poweroff samples are in boards:
```
$ find -name "*.c" | xargs grep --color sys_poweroff
./boards/esp32/deep_sleep/src/main.c:   sys_poweroff();
./boards/mimxrt595_evk/system_off/src/main.c:   sys_poweroff();
./boards/nrf/system_off/src/main.c:     sys_poweroff();
./boards/numaker/system_off/src/main.c: sys_poweroff();
./boards/stm32/power_mgmt/standby_shutdown/src/main.c:          sys_poweroff();
./boards/stm32/power_mgmt/stm32wb_ble/src/main.c:       sys_poweroff();
./boards/stm32/power_mgmt/wkup_pins/src/main.c: sys_poweroff();
./boards/ti/cc13x2_cc26x2/system_off/src/main.c:        sys_poweroff();

```

Reply:
Usually this is because enabling the wakeup source is board/soc specific (we still don't have a generic way of doing this in-tree), however, this PR enables the RTC wakeup source from within the sys_poweroff implementation, so it can be generic :)

Reply:
Thanks of your advice.
After review, there are some pins relative to board and decide to disable these specific pins to save power in power down mode.

Reply:
ok, I guess it's not generic anymore :)